dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/kcp/CompilerExtensionConfiguration.kt

```diff
@@ -0,0 +1,45 @@
+package com.datadog.gradle.plugin.kcp
+
+import com.datadog.gradle.plugin.InstrumentationMode
+
+/**
+ * Configuration for the Kotlin compiler plugin extension.
+ *
+ * This class defines settings that control how the compiler plugin applies instrumentation
+ * for various features in the Gradle plugin configuration.
+ */
+class CompilerExtensionConfiguration {
```

Comment:
maybe call it just `InstrumentationConfiguration`? We don't need `Compiler` in the name, it is technical details.